### PR TITLE
Take standard parameters of URL query string into account

### PR DESCRIPTION
**Pull request by esaracco**
Update the hydrate() method to take into account the parameters that do not have any pattern in the URL.

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Hello Emmanuel,

Thanks for your PR. Please make sure the test suite and linter complete (`npm run pre-commit`).

**Comment by esaracco** [comment]
Hello Nicolas,

Yes, sorry. Done!

File: template/src/main.js [hunk 1]
```diff
@@ -424,6 +424,11 @@ function init () {
       });
     }
 
+    // selects
```
Comment (NicolasCARPi) [inline template/src/main.js:427]: is this really needed?

**Review by NicolasCARPi — CHANGES_REQUESTED** [review]
So your changes made me realize that eslint was not reading the `.mjs` files. So I pushed a new commit on `dev` branch to fix that and now you need to adjust your new code.

Also I believe it would be a good idea to have an example of such URL in the example folder.

**Comment by esaracco** [comment]
@NicolasCARPi It should be fixed now.

File: template/src/main.js (hunk 1 above)
Comment (esaracco) [reply template/src/main.js]: I removed it.

File: template/index.html [hunk 2]
```diff
@@ -383,6 +383,16 @@ <h3>{{__ "Parameters"}}</h3>
               <div class="form-group">
                 <label class="col-md-3 control-label" for="sample-request-param-field-{{field}}">{{field}}</label>
                 <div class="input-group">
+                  {{#if allowedValues}}
+                  <div class="input-group-addon sample-request-select">
+                    <select>
+                      <option value=""></option>
```
Comment (NicolasCARPi) [inline template/index.html:389]: Why put an empty option here? If we must select from a list of allowed values, this is not needed.

Comment (esaracco) [reply template/index.html:389]: @NicolasCARPi It make sense if the parameter is optional.

**Comment by esaracco** [comment]
> Also, I'm thinking that the text input on the right of the select should not be here: 2021-10-01-180724_399x66_scrot

@NicolasCARPi It seemed to me that this was consistent with the other parameters displayed. I found that clearer. But I can hide the input if you want.

**Comment by NicolasCARPi** [comment]
I believe it makes more sense that if you have a list of allowed values, you can only select from that list and not have a "free text" input next to it. Also it would make the UI clearer.

File: template/index.html (hunk 2 above)
Comment (NicolasCARPi) [reply template/index.html:389]: I see, then add it after the loop on values, and maybe instead of being empty, name it  `null` to make it a bit more clear that this means we don't send anything with this parameter.

Comment (esaracco) [reply template/index.html]: Done.

**Comment by esaracco** [comment]
> I believe it makes more sense that if you have a list of allowed values, you can only select from that list and not have a "free text" input next to it. Also it would make the UI clearer.

@NicolasCARPi Done.

File: package.json [hunk 3]
```diff
@@ -42,6 +42,7 @@
     "win32"
   ],
   "dependencies": {
+    "apidoc": "^0.29.0",
```
Comment (NicolasCARPi) [inline package.json:45]: This is an error, right?

File: template/src/css/main.css [hunk 4]
```diff
@@ -33,14 +33,24 @@
 }
 
 .border-danger {
-  border: 1px solid var(--red);
+  border: 1px solid var(--red) !important;
```
Comment (NicolasCARPi) [inline template/src/css/main.css:36]: why is this needed now? I'd preferer if css doesn't contain any `!important` as it's a sign of incorrect cascade.

File: package.json (hunk 3 above)
Comment (esaracco) [reply package.json]: Fixed

File: template/src/css/main.css (hunk 4 above)
Comment (esaracco) [reply template/src/css/main.css]: Fixed

**Comment by esaracco** [comment]
Hello @NicolasCARPi,
I wrote a context menu so that the authorized values are displayed when clicking in the textarea of the "Body" section in "json" mode.
Can I push it into this pull request so that we can discuss it?

popup-menu-allowed-values

**Comment by NicolasCARPi** [comment]
Yes ok for the label and font sizes!

**Comment by esaracco** [comment]
> Yes ok for the label and font sizes!

@NicolasCARPi Done!

File: template/src/css/main.css [hunk 5]
```diff
@@ -112,6 +112,19 @@ td.code {
   user-select: none;
 }
 
+.label-required {
```
Comment (NicolasCARPi) [inline template/src/css/main.css:115]: I think it would be better to have a class for labels, and then optional/required which is just changing the color

File: template/src/css/main.css [hunk 6]
```diff
@@ -105,18 +105,18 @@ td.code {
...
+.label {
   float: right;
-  background-color: grey;
   margin-top: 4px;
   user-select: none;
 }
 
-.label-required {
-  float: right;
+.label.label-optional {
```
[the remark this answers is not in the document]
Comment (NicolasCARPi) [reply template/src/css/main.css:114]: My bad, I wrote it too fast, there is no child of label! So I would suggest simply removing the `label-` part, so the selector becomes `.label.optional`, avoiding a redundant `label` word.

[the remark this answers is not in the document]
Comment (esaracco) [reply template/src/css/main.css]: Done.
